File: paplj-eclipse/plugin/src/main/java/com/virtlink/paplj/PapljModule.java
package com.virtlink.paplj;

import com.google.inject.assistedinject.FactoryModuleBuilder;
import com.virtlink.aesi.eclipse.editors.AesiSourceViewerConfiguration;
import com.virtlink.aesi.eclipse.editors.ColorManager;
import com.virtlink.aesi.eclipse.editors.IContentAssistProcessorFactory;
import com.virtlink.editorservices.codecompletion.ICodeCompleter;
import com.virtlink.editorservices.eclipse.AesiModule;
import com.virtlink.paplj.codecompletion.DummyCodeCompleter;

public class PapljModule extends AesiModule {
	@Override
	protected void bindCodeCompletion() {
		bind(ICodeCompleter.class).to(DummyCodeCompleter.class);
		// TODO: Add other dependencies
		super.bindCodeCompletion();
	}
}

File: paplj-lsp/src/main/kotlin/com/virtlink/paplj/lsp/PapljModule.kt
package com.virtlink.paplj.lsp

import com.virtlink.editorservices.codecompletion.ICodeCompleter
import com.virtlink.editorservices.lsp.AesiModule
import com.virtlink.paplj.codecompletion.DummyCodeCompleter

class PapljModule : AesiModule() {

    override fun configure() {
        super.configure()
        bind(ICodeCompleter::class.java).to(DummyCodeCompleter::class.java)
    }
}